indexing-service/src/main/java/org/apache/druid/indexing/firehose/IngestSegmentFirehoseFactory.java

```diff
@@ -216,9 +216,7 @@ public Firehose connect(InputRowParser inputRowParser, File temporaryDirectory)
       for (TimelineObjectHolder<String, DataSegment> holder : timeLineSegments) {
         for (PartitionChunk<DataSegment> chunk : holder.getObject()) {
           final DataSegment segment = chunk.getObject();
-          if (!segmentFileMap.containsKey(segment)) {
-            segmentFileMap.put(segment, segmentLoader.getSegmentFiles(segment));
-          }
+          segmentFileMap.putIfAbsent(segment, segmentLoader.getSegmentFiles(segment));
```

Comment:
This needs to be `computeIfAbsent`, since there are side effects.

Reply:
Done.

Reply:
Done.